File: dailyQuestion/2020/2020-07/07-31/golang/solution_sum.go
package main

import "fmt"

func missingTwo(nums []int) []int {
	n := len(nums) + 2
	sum := 0
	for _, num := range nums {
		sum += num
	}
	sumTwo := n*(n+1)/2 - sum
	limit := sumTwo / 2
	sum = 0
	for _, num := range nums {
		if num <= limit {
			sum += num
		}
	}
	one := limit*(limit+1)/2 - sum
	return []int{one, sumTwo - one}

}

func main() {
	nums := []int{3, 4, 5, 1}
	result := missingTwo(nums)
	fmt.Println(result)
}

File: dailyQuestion/2020/2020-07/07-31/python/solution_sum.py
def missingTwo(nums: [int]) -> [int]:
    n = len(nums) + 2
    sum_nums = 0
    for num in nums:
        sum_nums += num
    sumTwo = int((n * (n + 1) / 2) - sum_nums)
    limit = sumTwo >> 1
    sum_nums = 0
    for num in nums:
        if num <= limit:
            sum_nums += num
    one = int(limit * (limit + 1) / 2 - sum_nums)
    return [one, sumTwo - one]

if __name__ == "__main__" :
    nums = [2,3]
    result = missingTwo(nums)
    print(result)
